app/src/main/kotlin/com/simplemobiletools/dialer/activities/SettingsActivity.kt

```diff
@@ -261,4 +290,52 @@ class SettingsActivity : SimpleActivity() {
             config.alwaysShowFullscreen = settings_always_show_fullscreen.isChecked
         }
     }
+
+    private fun setupCallsImportExport() {
+        settings_calls_import_holder.setOnClickListener {
+            getContent.launch(callHistoryFileType)
+        }
+
+        settings_calls_export_holder.setOnClickListener {
+            val fileName = "call_history_${getCurrentFormattedDateTime()}"
+            saveDocument.launch(fileName)
+        }
+    }
+
+
+    private fun importCallHistory(uri: Uri) {
+        runCatching {
+            val jsonString = contentResolver.openInputStream(uri)!!.use { inputStream ->
+                inputStream.bufferedReader().readText()
+            }
+
+            val recentsType = object : TypeToken<List<RecentCall>>() {}.type
+            val objects = Gson().fromJson<List<RecentCall>>(jsonString, recentsType)
+
+            RecentsHelper(this).restoreRecentCalls(this, objects) {
+                toast(R.string.importing_successful)
+            }
+        }.onFailure {
+            toast(R.string.importing_failed)
+        }
+    }
+
+    private fun exportCallHistory(recents: ArrayList<RecentCall>, uri: Uri) {
+        if (recents.isEmpty()) {
+            toast(R.string.no_entries_for_exporting)
+        } else {
+            runCatching {
```

Comment:
avoid such runCaching please, we arent using it anywher and it is hard to read

Reply:
Do you mean that we use the usual try/catch instead?

Reply:
sure